### ISSUE TITLE
Build issues on CentOS6

### ISSUE THREAD

**Issue by chaosphere2112** [UV-CDAT/uvcdat#1519]
I'm testing the current master's build on a clean CentOS6 VM (doing the footwork towards getting a non-lean RPM built for the 2.3 release), and I'm having a couple of issues with the build/configure process.

1. `cmake` can't find my qt install (which is from the binary we provide), installed at /usr/local/uvcdat/qt/4.8.4. Easy enough to work around (`-DQT_QMAKE_EXECUTABLE="/usr/local/uvcdat/qt/4.8.4/bin/qmake"`), but probably should be addressed. The alternative workaround (`export PATH="$PATH:/usr/local/uvcdat/qt/4.8.4/bin"`) also works. Neither is ideal, imo. Since we are telling everyone on redhat/centOS to use our Qt, can we have CMake actually look in that location by default?
2. VTK is failing to build because `cmake` isn't finding the OpenGL headers.

```
loading initial cache file
  /home/sam/Projects/build/VTK-prefix/tmp/VTK-cache.cmake

  CMake Error at
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:108 (message):

    Could NOT find OpenGL (missing: OPENGL_gl_LIBRARY OPENGL_INCLUDE_DIR)

  Call Stack (most recent call first):

    /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:315 (_FPHSA_FAILURE_MESSAGE)
    /usr/share/cmake/Modules/FindOpenGL.cmake:161 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
    Rendering/OpenGL/CMakeLists.txt:199 (find_package)

  

  

  CMake Error: The following variables are used in this project, but they are
  set to NOTFOUND.

  Please set them or make sure they are set and tested correctly in the CMake
  files:

  OPENGL_INCLUDE_DIR (ADVANCED)

     used as include directory in directory /home/sam/Projects/build/build/VTK/Geovis/Core

```

I tried hacking around this one by giving it a -DOPENGL_INCLUDE_DIR="/usr/include/GL", but I got the exact same message.

I added a [package list to the wiki](https://github.com/UV-CDAT/uvcdat/wiki/System-Requirements#redhatcentos) for the RedHat/CentOS equivalents of the Ubuntu packages, because I was tired of trying to figure them out every time. If anything looks like it's missing, please let me know.

**Comment by doutriaux1** [comment]
thanks @chaosphere2112, we copuld add /usr/local/blah to the CMakeLists.txt, and/or we could update the build instructions.

**Comment by aashish24** [comment]
@doutriaux1 please no. The issue is that on this machine the GL headers are not installed properly or something similar.

**Comment by aashish24** [comment]
@chaosphere2112 CMake looks for qmake for various QT paths. CMake should not look into specific locations for Qt because there could be many version of Qt exists on the system and then which one your pick is not obvious. Also, the location where Qt is installed changes from OS-to-OS and we try to solve for RedHat is not ideal to me. But if you still insist, we can set a default search for Qmake under /usr and let's it pick the first one it sees.

**Comment by doutriaux1** [comment]
@aashish24 the point that @chaosphere2112 is making is that WE ask the user to use OUR Qt under RH6, so it would make sense that WE look there too when we build.

**Comment by aashish24** [comment]
@doutriaux1 yup. I know. Please see my earlier comment.

**Comment by doutriaux1** [comment]
exactly, but it's not a priority, I put 3.0 milestone, but if one of us ( @chaosphere2112 ?) cares enough and has time, it could make it into 2.4
@chaosphere2112 are we going to use Qt 4.8 for the updated GUI? Or Qt 5?

**Comment by aashish24** [comment]
I think we probably have to stick with Qt 4.8 for a while but it would be nice to move to Qt5.

**Comment by chaosphere2112** [comment]
We'll still be using 4.8; PySide's Qt5 support is still a ways out.

**Comment by chaosphere2112** [comment]
@aashish24 @doutriaux1 As for the Qt thing, we could just change where our tar of Qt installs it; putting it somewhere that CMake will actually pick up. I'm fine coming at it from either side.

I left the laptop with this VM at home today, so I'll not be able to do anything on it till tomorrow. As for GL not being properly installed, it's just a vanilla CentOS6 install, so I'm not sure what would have happened.

**Comment by aashish24** [comment]
@chaosphere2112 GL is not installed in minimal builds. I will see if I can send you the list of packages you need to install (if you have not installed them already?)

**Comment by durack1** [comment]
@aashish24 rather than sending an email, want to update the CentOS6 dependency list at https://github.com/UV-CDAT/uvcdat/wiki/System-Requirements

**Comment by aashish24** [comment]
@durack1 :+1:  that's what I was planning to do. Add it there and send an email with the link.

**Comment by doutriaux1** [comment]
@chaosphere2112 where did you update the requirements? You mentioned you already did this.

**Comment by chaosphere2112** [comment]
@doutriaux1 It's down on the bottom of the system requirements page (https://github.com/UV-CDAT/uvcdat/wiki/System-Requirements#centos)
